pkg/connector/spaces.go

```diff
@@ -109,103 +112,197 @@ func (o *spaceBuilder) Grants(
 	annotations.Annotations,
 	error,
 ) {
-	permissionsList, ratelimitData, err := o.client.GetSpacePermissions(
-		ctx,
-		resource.Id.Resource,
-	)
-	outputAnnotations := WithRateLimitAnnotations(ratelimitData)
-	if err != nil {
-		return nil, "", outputAnnotations, err
+	spaceKey := resource.Id.Resource
+	if len(o.SpacePermissionsCache[spaceKey]) == 0 {
```

Comment:
in memory caches attached to the builder need a mutex. This connector is generally used on prem, but that could change, and we never want to accidentally leak data across tenants.